collector/scripts/twitter.py

```diff
@@ -0,0 +1,95 @@
+        consumer_key, consumer_secret, access_token, access_token_secret
+    )
+
+    # Instantiate the tweepy API
+    api = tweepy.API(auth, wait_on_rate_limit=True)
+
+    try:
+        if (searchQuery == ''):
+            tweets = api.user_timeline(screen_name=username, tweet_mode = 'extended', count=tweetCount)
+        else:
+            tweets = api.search_tweets(q=searchQuery, tweet_mode = 'extended', count=tweetCount)
+
+        # Pulling Some attributes from the tweet
+        attributes_container = [
+            [tweet.id, tweet.user.screen_name, tweet.created_at, tweet.favorite_count, tweet.source, tweet.full_text]
+            for tweet in tweets
+        ]
+        
+        # Creation of column list to rename the columns in the dataframe
+        columns = ["id", "Screen Name", "Date Created", "Number of Likes", "Source of Tweet", "Tweet"]
+
+        # Creation of Dataframe
+        tweets_df = pd.DataFrame(attributes_container, columns=columns)
+
+        totalTokens = 0
+        for index, row in tweets_df.iterrows():
+            meta_link = twitter_meta(row, True)
+            output_filename = f"twitter-{username}-{row['Date Created']}.json"
+            output_path = f"./outputs/twitter-logs"
+
+            transaction_output_filename = f"tweet-{username}-{row['id']}.json"
+            transaction_output_dir = f"../server/storage/documents/twitter-{username}"
+
+            if not os.path.isdir(output_path):
+                os.makedirs(output_path)
+
+            if not os.path.isdir(transaction_output_dir):
+                os.makedirs(transaction_output_dir)
+
+            full_text = twitter_meta(row)
+            tokenCount = len(tokenize(full_text))
+            meta_link['pageContent'] = full_text
+            meta_link['token_count_estimate'] = tokenCount
+            totalTokens += tokenCount
+
+            with open(f"{output_path}/{output_filename}", 'w', encoding='utf-8') as file:
+                json.dump(meta_link, file, ensure_ascii=True, indent=4)
+
+            with open(f"{transaction_output_dir}/{transaction_output_filename}", 'w', encoding='utf-8') as file:
+                json.dump(meta_link, file, ensure_ascii=True, indent=4)
+            
+            # print(f"{transaction_output_dir}/{transaction_output_filename}")
+
+        print(f"{tokenCount} tokens written over {tweets_df.shape[0]} records.")
+
+    except BaseException as e:
+        print("Status Failed: ", str(e))
+        time.sleep(3)
+
+
+def twitter_meta(row, metadata_only = False):
+  url = f"http://twitter.com/anyuser/status/{row['id']}"
+  title = f"Tweet {row['id']}"
+  meta = {
+    'url': url,
+    'title':  title,
+    'description': 'Tweet from ' + row["Screen Name"],
+    'published': row["Date Created"].strftime('%Y-%m-%d %H:%M:%S'),
+    'wordCount': len(row["Tweet"]),
+  }
+  return "Tweet JSON Metadata:\n"+json.dumps(meta)+"\n\n\nText Content:\n" + row["Tweet"] if metadata_only == False else meta
```

Comment:
I am fairly sure this function does not work because `http://twitter.com/anyuser/status/{row['id']}` will try to resolve the `@anyuser` handle doing this - which won't work since the users handle can be anything.

Reply:
This does work since I’ve used it to scrape. This is a known workaround in Twitter; it’ll redirect to the right user ID you know the i I d.

Reply:
Ah, good to know. Had no idea!